src/js/api/authentication.mjs

```
@@ -0,0 +1,25 @@
+import { request } from "./request.mjs";
+
+// TODO: Document function
+export async function login(email, password) {
+    try {
+        const data = { email: email, password: password };
+        const response = await request("/social/auth/login", "POST", data);
+        const json = await response.json();
+        return { success: response.ok, data: json };
+    } catch (error) {
+        throw new Error(error);
+    }
+}
+
+// TODO: Document function
+export async function register(name, email, password) {
+    try {
+        const data = { name: name, email: email, password: password };
+        const response = await request("/social/auth/register", "POST", data);
+        const json = await response.json();
+        return { success: response.ok, data: json };
+    } catch (error) {
+        throw new Error(error);
+    }
+}
```

A couple of things are missing and wrong here:

1) A `try/catch` was supposed to be here.

2) A conditional statement to check if the response is OK.

3) `catch block`: to suppress an error
     `throw`: to create an error

     Understand the differences, they have different purposes. Here you're basically catching an error and 
     throwing it again.

    **Suggestion:** remove the throw from the catch block and use the try/catch block in a UI/UX code block 
    e.g event listener.